src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/FlowStateAllowedEventService.java

```diff
@@ -493,29 +493,25 @@ public boolean isAllowedOnState(String stateFullName, CaseEvent caseEvent) {
     }
 
     public boolean isAllowedOnStateForSpec(String stateFullName, CaseEvent caseEvent) {
-
         return ALLOWED_EVENTS_ON_FLOW_STATE_SPEC
             .getOrDefault(stateFullName, emptyList())
             .contains(caseEvent);
-
     }
 
     public boolean isAllowed(CaseDetails caseDetails, CaseEvent caseEvent) {
         CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseDetails);
 
         if ((caseData.getSuperClaimType() != null && caseData.getSuperClaimType().equals(SPEC_CLAIM))
             || caseEvent.toString().equals("CREATE_CLAIM_SPEC")) {
```

Comment:
CaseEvent is an enum. Instead of toString.equals("..."), you can use caseEvent.equals(CaseEvent.CREATE_CLAIM_SPEC) and that way, if the name of that CaseEvent value changes, the compiler will warn. Comparing with a String, it won't.
Also, since enum values are automatically a shared instance, you could use caseEvent == CaseEvent.CREATE_CLAIM_SPEC and it won't throw null pointer exception even if caseEvent is null. 
The null risk can also be removed by reversing the equals to CaseEvent.CREATE_CLAIM_SPEC.equals(caseEvent)

And the same happens with the previous line, which could be SPEC_CLAIM.equals(caseData.superClaimType) instead of an and